lottery_categories/router.py

```
@@ -0,0 +1,114 @@
+        if not success:
+            # This might be because nothing changed, or DB error not caught by PyMongoError
+            # Re-fetch to see if it's a "no change" or actual error
+            updated_category_check = categories_db.get_category_by_id(category_id)
+            if updated_category_check and \
+               all(getattr(updated_category_check, k, None) == v for k, v in category_update.model_dump(exclude_unset=True).items() if v is not None):
+                return updated_category_check # No actual change, but data is consistent
+            raise HTTPException(status_code=500, detail="Failed to update lottery category or no changes made.")
+
+        updated_category = categories_db.get_category_by_id(category_id)
+        if not updated_category: # Should not happen if update was successful
+             raise HTTPException(status_code=500, detail="Failed to retrieve category after update.")
```

Flake8 (7.2.0)

[error] indentation is not a multiple of 4

(E111)